python_modules/libraries/dagster-snowflake-pyspark/dagster_snowflake_pyspark_tests/test_get_snowflake_options.py

```diff
@@ -0,0 +1,49 @@
+from typing import Dict
+
+import pytest
+from dagster._check import CheckError
+from dagster._core.storage.db_io_manager import TableSlice
+from dagster_snowflake_pyspark.csid import SNOWFLAKE_PARTNER_CONNECTION_IDENTIFIER_PYSPARK
+from dagster_snowflake_pyspark.snowflake_pyspark_type_handler import _get_snowflake_options
+
+
+@pytest.fixture
+def snowflake_config() -> Dict[str, str]:
+    return {
+        "account": "account",
+        "user": "user",
+        "password": "password",
+        "database": "database",
+        "warehouse": "warehouse",
+    }
+
+
+@pytest.fixture
+def table_slice() -> TableSlice:
+    return TableSlice("table_name", "schema_name", "database_name")
+
+
+def test_get_snowflake_options(snowflake_config, table_slice):
+    options = _get_snowflake_options(snowflake_config, table_slice)
+    assert options == {
+        "sfURL": "account.snowflakecomputing.com",
+        "sfUser": "user",
+        "sfPassword": "password",
+        "sfDatabase": "database",
+        "sfWarehouse": "warehouse",
+        "sfSchema": "schema_name",
+        "APPLICATION": SNOWFLAKE_PARTNER_CONNECTION_IDENTIFIER_PYSPARK,
+    }
+
+
+def test_missing_warehouse(snowflake_config, table_slice):
+    del snowflake_config["warehouse"]
+
+    with pytest.raises(CheckError):
+        _get_snowflake_options(snowflake_config, table_slice)
+
+
+def test_override_snowflake_partner_connection_id(snowflake_config, table_slice):
+    snowflake_config["application"] = "foobar"
+    options = _get_snowflake_options(snowflake_config, table_slice)
+    assert options.get("APPLICATION") == "foobar"
```

Comment:
The test should fail if `APPLICATION` is not in options, so we should avoid using `get()`. Optionally, your could test that the key is in the dictionary.

```suggestion
    assert "APPLICATION" in options  # this is optional
    assert options["APPLICATION"] == "foobar"
```

Reply:
I don't understand why that's a problem here.

If application is not in the options dict, then `options.get("APPLICATION")` will evaluate to `None` and the test will fail with `None != "foobar"`.

The only difference would be that `options["APPLICATION"]` will throw an KeyError instead.